roqoqo-iqm/src/backend.rs

```diff
@@ -0,0 +1,459 @@
+                            });
+                        }
+                    } else if let Ok(inner_op) = TwoQubitOperation::try_from(op) {
+                        if self
+                            .device
+                            .two_qubit_gate_time(
+                                inner_op.hqslang(),
+                                inner_op.control(),
+                                inner_op.target(),
+                            )
+                            .is_none()
+                        {
+                            return Err(RoqoqoBackendError::OperationNotInBackend {
+                                backend: "IQM",
+                                hqslang: inner_op.hqslang(),
+                            });
+                        }
+                    } else if let Ok(inner_op) = MultiQubitOperation::try_from(op) {
+                        if self
+                            .device
+                            .multi_qubit_gate_time(inner_op.hqslang(), inner_op.qubits())
+                            .is_none()
+                        {
+                            return Err(RoqoqoBackendError::OperationNotInBackend {
+                                backend: "IQM",
+                                hqslang: inner_op.hqslang(),
+                            });
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Query results of a submitted job.
+    ///
+    /// # Arguments
+    ///
+    /// `id` - The job ID for the query
+    ///
+    /// # Returns
+    ///
+    /// `Ok(IqmRunResult)` - Result of the job (status can be pending)
+    /// `Err(RoqoqoBackendError)` - If something goes wrong with HTML requests or response is not
+    /// formatted correctly
+    pub fn get_results(&self, id: &str) -> Result<IqmRunResult, RoqoqoBackendError> {
+        let client = reqwest::blocking::Client::builder()
+            .https_only(true)
+            .build()
+            .map_err(|x| RoqoqoBackendError::NetworkError {
+                msg: format!("could not create https client {:?}", x),
+            })?;
+
+        let job_url = self.device.remote_host().to_owned() + "/" + id;
+
+        let result = client
+            .get(job_url)
+            .headers(_construct_headers(&self.access_token))
+            .send()
+            .map_err(|e| RoqoqoBackendError::NetworkError {
+                msg: format!("Error during GET request: {:?}", e),
+            })?;
+
+        let iqm_result: IqmRunResult =
+            result
+                .json::<IqmRunResult>()
+                .map_err(|e| RoqoqoBackendError::NetworkError {
+                    msg: format!("Could not convert result into IqmRunResult: {:?}", e),
+                })?;
+
+        // TODO print warnings if any
+        // if iqm_result.warnings.is_some() {
+        //     println!("Warnings: {:?}", iqm_result.warnings.unwrap());
+        // }
```

Comment:
Any reason this is commented out? This looks good to me, although we maybe should use eprintln instead of println